tests/test_integration/test_utils/test_cdf.py

```diff
@@ -0,0 +1,56 @@
+from collections import Counter
+
+import pytest
+from cognite.client.data_classes import AssetList, AssetWrite, AssetWriteList
+
+from cognite_toolkit._cdf_tk.client import ToolkitClient
+from cognite_toolkit._cdf_tk.utils.cdf import metadata_key_counts
+
+
+@pytest.fixture(scope="session")
+def small_hierarchy(toolkit_client: ToolkitClient) -> AssetList:
+    assets = AssetWriteList(
+        [
+            AssetWrite(
+                name="Root Asset",
+                external_id="toolkit_test_metadata_key_counts_root_asset",
+                metadata={"metadata_key_count_test": "does-not-matter"},
+            ),
+            AssetWrite(
+                name="Child Asset",
+                parent_external_id="toolkit_test_metadata_key_counts_root_asset",
+                external_id="toolkit_test_metadata_key_counts_child_asset",
+                metadata={
+                    "metadata_key_count_test": "does not matter either",
+                    "metadata_key_count_test_2": "does not matter either",
+                },
+            ),
+        ]
+    )
+    existing = toolkit_client.assets.retrieve_multiple(external_ids=assets.as_external_ids(), ignore_unknown_ids=True)
+    if missing := [asset for asset in assets if asset.external_id not in set(existing.as_external_ids())]:
+        created = toolkit_client.assets.create(missing)
+        existing.extend(created)
+    return existing
+
+
+class TestMetadataKeyCounts:
```

Comment:
no testing for datasets (single vs. multiple), multiple hierarchies, combination of datasets and hierarchies